src/app/collective-rewards/rewards/builders/EstimatedRewards.tsx

```diff
@@ -47,22 +46,33 @@ const TokenRewards: FC<TokenRewardsProps> = ({ builder, gauge, token: { id, symb
     data: totalPotentialRewards,
     isLoading: totalPotentialRewardsLoading,
     error: totalPotentialRewardsError,
-  } = useGetTotalPotentialReward()
+  } = useReadBackersManager({
+    functionName: 'totalPotentialReward',
+  })
   const {
     data: rewardShares,
     isLoading: rewardSharesLoading,
     error: rewardSharesError,
-  } = useGetRewardShares(gauge)
+  } = useReadGauges({ addresses: [gauge], functionName: 'rewardShares' })
   const {
     data: { cycleNext },
     isLoading: cycleLoading,
     error: cycleError,
   } = useCycleContext()
   const {
-    data: backerRewardsPct,
+    data: rawBackerRewardsPct,
     isLoading: backerRewardsPctLoading,
     error: backerRewardsPctError,
-  } = useGetBackerRewardPercentage(builder, cycleNext.toSeconds())
+  } = useReadBuilderRegistry({
+    functionName: 'backerRewardPercentage',
+    args: [builder],
+  })
+
+  const backerRewardsPct = useMemo(() => {
+    const [previous, next, cooldownEndTime] = rawBackerRewardsPct ?? [0n, 0n, 0n]
+
+    return getBackerRewardPercentage(previous, next, cooldownEndTime, cycleNext.toSeconds())
+  }, [rawBackerRewardsPct, cycleNext])
```

Comment:
NIT: Probably in this case it made sense to have the hook as the operation isn't atomic

Reply:
I disagree. The addition to the component is minimal and completely related to itself.